src/python/pants/backend/python/rules/run_setup_py.py

```diff
@@ -152,12 +156,83 @@ class SetupPyChroot:
   setup_keywords_json: str
 
 
+@dataclass(frozen=True)
+class RunSetupPyRequest:
+  """A request to run a setup.py command."""
+  exported_target: ExportedTarget
+  chroot: SetupPyChroot
+  args: Tuple[str, ...]
+
+
+@dataclass(frozen=True)
+class RunSetupPyResult:
+  """The result of running a setup.py command."""
+  output: Digest  # The state of the chroot after running setup.py.
+
+
 @dataclass(frozen=True)
 class SetuptoolsSetup:
   """The setuptools tool."""
   requirements_pex: Pex
 
 
+class SetupPyOptions(GoalSubsystem):
+  name = "setup-py2"
+
+  @classmethod
+  def register_options(cls, register):
+    super().register_options(register)
+    register(
+      '--args', type=list, member_type=str,
+      help="Arguments to pass directly to setup.py, e.g. "
+           "`--setup-py2-args=\"bdist_wheel\"`. If unspecified, we just dump the setup.py chroot."
```

Comment:
Even better would be saying `--setup-py2-args='bdist_wheel --python-tag py36.py37'`. The reason I recommend this is to clarify that the string gets shlexed.